packages/esm-sms-app/src/hooks/useLogs.ts

```diff
@@ -1,24 +1,36 @@
 import useSWRInfinite from 'swr/infinite';
-import { openmrsFetch, useConfig, type FetchResponse } from '@openmrs/esm-framework';
-import { type SMSLog, type SMSLogsResponse } from '../types';
-import { type ConfigObject } from '../config-schema';
+import { openmrsFetch, type FetchResponse } from '@openmrs/esm-framework';
+import { type SMSLogsResponse } from '../types';
 
-export function useLogsRecords() {
+export function useSMSLogs(
+  pageNumber: number,
+  pageSize: number,
+  phoneNumber: string = '',
+  sortColumn: string = 'id',
+  sortDirection: string = 'desc',
+) {
   const getKey = (pageIndex, previousPageData) => {
     if (previousPageData && !previousPageData.rows.length) return null;
 
-    return `/ws/sms/log?page=${pageIndex + 1}`;
+    let url = `/ws/sms/log?page=${pageNumber}&rows=${pageSize}&sortColumn=${sortColumn}&sortDirection=${sortDirection}`;
+
+    if (phoneNumber.trim() !== '') {
+      url += `&phoneNumber=${encodeURIComponent(phoneNumber)}`;
+    }
+
+    return url;
   };
 
-  const { data, error, size, setSize, isValidating, isLoading, mutate } = useSWRInfinite<
+  const { data, error, setSize, isValidating, isLoading, mutate } = useSWRInfinite<
     FetchResponse<SMSLogsResponse>,
     Error
   >(getKey, openmrsFetch);
 
-  const smsLogs: SMSLog[] = data ? [].concat(...data.map((page) => page.data.rows)) : [];
+  const responseData = data ? data[0].data : null;
```

Comment:
Why are we only using the first set of data, and in case we are, why are we using `SWRInfinite` then?

Reply:
corrected